### PR TITLE
Logging as NPM module

### LINKED ISSUE #899: Refactor logging into a separate NPM module
As requested and discussed with @pavlovcik .

### PR DESCRIPTION
**Pull request by gitcoindev**
Resolves https://github.com/ubiquity/ubiquibot/issues/899

### CODE REVIEW

**Comment by gitcoindev** [comment]
Hi @pavlovcik the dependencies looks fine, NPM package builds. I need still to test and publish from my fork though and see if it can be correctly imported and built, therefore I set this pull request still as a draft.

**Comment by gitcoindev** [comment]
@pavlovcik pull request ready for the review.

Package correctly exported:

```
npm notice 
npm notice 📦  ubiquibot-logger@0.3.0
npm notice === Tarball Contents === 
npm notice 49B    README.md              
npm notice 1.9kB  dist/logs.d.ts         
npm notice 12.5kB dist/logs.js           
npm notice 27.2kB dist/logs.js.map       
npm notice 604B   dist/pretty-logs.d.ts  
npm notice 4.2kB  dist/pretty-logs.js    
npm notice 9.3kB  dist/pretty-logs.js.map
npm notice 2.2kB  package.json           
npm notice === Tarball Details === 
npm notice name:          ubiquibot-logger                        
npm notice version:       0.3.0                                   
npm notice filename:      ubiquibot-logger-0.3.0.tgz              
npm notice package size:  11.0 kB                                 
npm notice unpacked size: 57.8 kB                                 
npm notice shasum:        ae296958daae3169bfde2866379216621c7c41f5
npm notice integrity:     sha512-F3BExAEPfQrHK[...]oECTw1tELcn0Q==
npm notice total files:   8                                       
npm notice 
npm notice Publishing to https://registry.npmjs.org/ with tag latest and default access
+ ubiquibot-logger@0.3.0
```

QA build with import from https://github.com/pavlovcik/ubiquibot/pull/60 builds without errors:


```
{refactor-general-with-npm-logging} ~/ubiquibot$ yarn build
yarn run v1.22.19
$ tsc
Done in 5.45s.
```

File: src/supabase/helpers/tables/pretty-logs.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  bgBlack = "\x1b[40m",
+  bgRed = "\x1b[41m",
+  bgGreen = "\x1b[42m",
+  bgYellow = "\x1b[43m",
+  bgBlue = "\x1b[44m",
+  bgMagenta = "\x1b[45m",
+  bgCyan = "\x1b[46m",
+  bgWhite = "\x1b[47m",
+}
+export enum LogLevel {
```
Comment (0x4007) [inline src/supabase/helpers/tables/pretty-logs.ts:179]: I'm impressed. It looks like you correctly took the latest code which I updated as recently as yesterday!

File: package.json [hunk 2]
```diff
@@ -1,35 +1,70 @@
...
+    "@supabase/supabase-js": "^2.4.0",
+    "dotenv": "^16.3.1",
+    "pino-std-serializers": "^6.2.2",
+    "probot": "^12.2.4"
   },
   "devDependencies": {
     "@commitlint/cli": "^18.4.3",
     "@commitlint/config-conventional": "^18.4.3",
-    "@types/node": "^20.10.0",
+    "@types/node": "^20.10.4",
```
Comment (0x4007) [inline package.json:67]: Why was this changed? Shouldn't we stay at LTS?

**Review by 0x4007 — APPROVED** [review]
Great work! I'm unsure how we can decouple the GitHub comments from this but the main reusable portion is 1. Supabase logging and 2. The nice colored and formatted outputs. 

My vision is to be able to reuse these for all of our projects, perhaps bundling with @ubiquity/ts-template

Logging `fatal` errors I expect will expedite triage for any fatal issues of our products. 

It will be nifty to integrate that with a telegram bot to post in a triage group chat for go live. Perhaps even automatically creating and funding issues on GitHub as well.

File: src/supabase/helpers/tables/pretty-logs.ts (hunk 1 above)
Comment (gitcoindev) [reply src/supabase/helpers/tables/pretty-logs.ts:179]: Yes, I took the latest code, glad to hear that!

File: package.json (hunk 2 above)
Comment (gitcoindev) [reply package.json:67]: Good catch, I think we can stay at ^20.10.0

Comment (0x4007) [reply package.json:67]: Okay you can address in a new pull request.

Comment (gitcoindev) [reply package.json:67]: @pavlovcik I did a test, changed to LTS `^20.10.0` and yarn installed 20.10.4 anyway, took the latest available patch version in `major.minor.patch` format:

+"@types/node@*", "@types/node@^20.10.0":
   version "20.10.4"
   resolved "https://registry.npmjs.org/@types/node/-/node-20.10.4.tgz"
   integrity sha512-D08YG6rr8X90YB56tSIuBaddy/UXAA9RKJoFvrsnogAum/0pmjkgi4+2nx9

Since node follows semantic versioning https://semver.org/ we can stay with `^20.10.4` as changing to `^20.10.0` will not have effect, the latest LTS patch version 20.10.x will be installed anyway.
